src/main/java/org/junit/runners/ParentRunner.java

```diff
@@ -206,6 +208,29 @@ private boolean areAllChildrenIgnored() {
     }
 
     /**
+     * Returns a {@link Statement}: notify all non-overridden {@code @Ignore} methods on this class
+     * and superclasses before executing {@code next}.
+     * @param next The next statement, or {@code null}, if there is no next statement.
+     * @param notifier The runner notifier.
+     */
+    protected Statement withIgnores(final Statement next, final RunNotifier notifier) {
+        return new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                for (final T each : getFilteredChildren()) {
```

Comment:
This sounds like a good use case for a collection library, i.e. a method like http://google-collections.googlecode.com/svn/trunk/javadoc/com/google/common/collect/Iterables.html#filter%28java.lang.Iterable%2C%20com.google.common.base.Predicate%29

Nevertheless the private implementation of runChildren() uses a private scheduler implementation, which do nothing else then creating own threads for each test method. These threads make sense for test methods, which are not ignored. The notification of ignored methods should run fast. Would you say, that the notified listeners can consume a lot of time, which perform better with threads?